DISCOVER/conf.py

```diff
@@ -3,6 +3,40 @@
 # If you wish to continue using _config.yml, make edits to that file and
 # re-generate this one.
 ###############################################################################
+
+version = '2.0'
+language = 'en'
+baseurl = 'https://discover-cookbook.numfocus.org'
+
+import json
+import os
+
+# Load language data from languages.json
+language_json_path = os.path.join(os.path.dirname(__file__), '_static', 'languages.json')
+language_data = []
+if os.path.exists(language_json_path):
+    with open(language_json_path, 'r', encoding='utf-8') as f:
+        all_languages = json.load(f)
+        # Filter out hidden languages
+        language_data = [lang for lang in all_languages if not lang.get('hidden', False)]
+
+# Create a mapping of language codes to their localized names
+language_names = {}
+for lang in language_data:
+    if "code" in lang and "name_local" in lang:
+        language_names[lang["code"]] = lang["name_local"]
+
+
+html_context = {
+    "languages": language_data,
+    "language_names": language_names,
```

Comment:
I think the logic for `language_names[current_language]` will break for hidden languages. The name should appear in the button, but not in the dropdown when we open it. Given we only even use `language_names[current_language]` and `current_language` is fixed within each build, I would define `current_language_name` as string instead of `language_names` dict. If we do that right before filtering hidden languages I think we'd get the desired behaviour.